### DISCUSSION TITLE
What are the best methods for running the agent on macOS?

### DISCUSSION THREAD

**Discussion by henrygd** [henrygd/beszel#276] [General]
Hello all,

I have very little experience with MacOS, so I don't know what to tell people when they ask for help.

I'm hoping I can get some tips and feedback from anyone here who is running the agent successfully.

Are you using launchd? Docker? Detached tmux session?

I would like to create a little guide with recommended approaches and examples.

Thanks!

**Comment by jianlongliu** [comment]
我也是遇到这个问题, 没几天用launchd重启服务临时解决
I also encountered this issue. Restarting the service with launchd temporarily resolved it.

**Comment by Zesty0wl** [comment]
I'm seeing this as well across multiple Macs. Intel and Apple Silicon. It's being throttled:

`Connection removed: IOHIDEventSystemConnection uuid:7120D30C-0611-47DF-B2C6-71A0E15739A6 pid:34263 process:beszel-agent type:Rate Controlled entitlements:0x0 caller:beszel-agent: beszel-agent + 474444 attributes:(null) state:0x1 events:0 mask:0x0 dropped:0 dropStatus:0 droppedMask:0x0 lastDroppedTime:NONE`

The log entry indicates that a connection associated with the beszel-agent process was removed by the macOS Input/Output HID Event System. The message suggests that the beszel-agent is making a large number of requests to the IOHID system, which has triggered rate-limiting or throttling mechanisms.

Here's an example from fs_usage. You can see that the agent is making a lot of read, write, and close operations in quick succession which is probably the root cause. We probably need a sleep statement in a loop somewhere.

14:05:16.144433  read              F=5    B=0x34
14:05:16.144505  read              F=5  [ 35]
14:05:16.144699  write             F=5    B=0x34
14:05:16.145519  read              F=5    B=0x34
14:05:16.145527  read              F=5  [ 35]
14:05:16.145564  write             F=5    B=0x24
14:05:16.145743  statfs64
14:05:16.146308  read              F=5    B=0x24
14:05:16.146315  read              F=5  [ 35]
14:05:16.146911  stat64                 [  2]
14:05:16.146915  stat64                 [  2]
14:05:16.146971  stat64
14:05:16.147221  open              F=6        (R__________X)
14:05:16.147227  fstat64           F=6
14:05:16.147230  fcntl             F=6   <GETFL>
14:05:16.147233  fcntl             F=6   <SETFL>
14:05:16.147238  fcntl             F=6   <GETFL>
14:05:16.147239  fcntl             F=6   <SETFL>
14:05:16.147252  fcntl             F=7   <SETFD>
14:05:16.147253  fcntl             F=8   <SETFD>
14:05:16.147253  fcntl             F=7   <GETFL>
14:05:16.147254  fcntl             F=7   <SETFL>
14:05:16.147257  fcntl             F=8   <GETFL>
14:05:16.147258  fcntl             F=8   <SETFL>
14:05:16.147268  fcntl             F=8   <GETFL>
14:05:16.147268  fcntl             F=8   <SETFL>
14:05:16.147269  fcntl             F=8   <GETFL>
14:05:16.147276  fcntl             F=9   <SETFD>
14:05:16.147276  fcntl             F=10  <SETFD>
14:05:16.148919  close             F=10
14:05:16.150101  read              F=9    B=0x0
14:05:16.150158  close             F=9
14:05:16.150397  close             F=6
14:05:16.150406  close             F=8
14:05:16.150481  read              F=7  [ 35]
14:05:16.156875  read              F=7    B=0x200
14:05:16.156900  read              F=7    B=0x200
14:05:16.156906  read              F=7    B=0x400
14:05:16.156919  read              F=7    B=0x800
14:05:16.156921  read              F=7    B=0x629
14:05:16.156922  read              F=7  [ 35]
14:05:16.157140  read              F=7    B=0x0
14:05:16.157151  close             F=7
14:05:16.245016  socket            F=6        <AF_UNIX, SOCK_STREAM, 0x0>
14:05:16.245027  fcntl             F=6   <SETFD>
14:05:16.245030  fcntl             F=6   <GETFL>
14:05:16.245036  fcntl             F=6   <SETFL>
14:05:16.245137  connect           F=6  [  2]
14:05:16.245292  close             F=6
14:05:16.245534  write             F=5    B=0x334
14:05:16.245564  write             F=5    B=0x34
14:05:16.245575  write             F=5    B=0x24
14:05:16.246843  read              F=5    B=0x24
14:05:16.246873  read              F=5  [ 35]

**Comment by Zesty0wl** [comment]
Actually, I took a chance and ran the agent with a nice 10 value, which seems to have been enough to place 'nice'.

To use this...

Download beszel agent
```shell
curl -sL "https://github.com/henrygd/beszel/releases/latest/download/beszel-agent_$(uname -s)_$(uname -m | sed -e 's/x86_64/amd64/' -e 's/armv6l/arm/' -e 's/armv7l/arm/' -e 's/aarch64/arm64/').tar.gz" | tar -xz -O beszel-agent | tee ./beszel-agent >/dev/null && chmod +x beszel-agent
``` 

Copy to /usr/local/bin
```shell
sudo cp beszel-agent /usr/local/bin
``` 
Create a LaunchDaemon
```shell
sudo nano /Library/LaunchDaemon/com.beszel.agent.plist
```

Edit the following, replacing the port and key as required
```shell
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
    <key>Label</key>
    <string>com.beszel.agent</string>

    <key>ProgramArguments</key>
    <array>
        <string>/bin/bash</string>
        <string>-c</string>
        <string>PORT=45876 KEY="ssh-ed25519 XXXXXX" /usr/local/bin/beszel-agent</string>
    </array>

    <key>RunAtLoad</key>
    <true/>

    <key>KeepAlive</key>
    <dict>
        <!-- Prevent excessive restarts -->
        <key>SuccessfulExit</key>
        <false/>
        <key>ThrottleInterval</key>
        <integer>60</integer> <!-- Minimum 60 seconds between restarts -->
    </dict>

    <key>StandardErrorPath</key>
    <string>/var/log/beszel-agent.err</string>
    <key>StandardOutPath</key>
    <string>/var/log/beszel-agent.log</string>

    <!-- Lower the agent's process priority -->
    <key>Nice</key>
    <integer>10</integer>
</dict>
</plist>
``` 

Load the LaunchDaemon
```shell
sudo launchctl load /Library/LaunchDaemon/com.beszel.agent.plist
```

Test if it's running
```shell
pgrep -l beszel-agent
38277 beszel-agent
```

**Reply by henrygd** [reply]
Thank you, this is extremely helpful! I really appreciate you sharing this!

Do you have a large number of temperature sensors by any chance? Or are you running a lot of Docker containers?

**Reply by cliss** [reply]
I'm just a random passerby, but this was extremely helpful; thank you, @Zesty0wl! :beers:

**Reply by Zesty0wl** [reply]
I left a couple of Mac Mini's running with nice set to 10 overnight and it's improved but not resolved, still very long gaps in reporting where the agent is not able to function due to being rate limited.

You could be on to something with the temp sensor idea. None of mine have many docker containers, maybe 8 at max. But they are reporting a lot of temp sensors...

Apple Silicon M2 Pro
<img width="695" alt="image" src="https://github.com/user-attachments/assets/65cd5802-2758-43fa-9788-6839bffabe61" />

**Reply by henrygd** [reply]
Can you please try setting the agent's `SENSORS` env var to an empty string (`SENSORS=""`)?

This will skip the sensor checks. If it runs smoothly after that then we'll have identified the issue and I can go about fixing it.

**Reply by Zesty0wl** [reply]
So, I made that change and it's been fine since.

<img width="1387" alt="image" src="https://github.com/user-attachments/assets/9463c725-db70-4084-8ffa-f030d273d1b5" />

**Reply by henrygd** [reply]
Great, thanks. I'll look further into it. Let me know if it starts happening again.

**Reply by Zesty0wl** [reply]
If I get 5 minutes I might take a look at the agent code too. My guess is that we just need a slight delay in the loop where we get sensor data. That's usually enough to get macOS to stop getting over excited.

**Reply by henrygd** [reply]
We're using gopsutil's sensors package to get the sensor data:

https://github.com/shirou/gopsutil/blob/master/sensors/sensors_darwin_arm64.go

I also wonder if this could be caused by purego / cross compilation. Maybe an [agent compiled natively on macos](https://beszel.dev/guide/compiling) would not have this issue.

**Reply by luxiangzh** [reply]
@henrygd Hi! I'm wondering if there're any plan to add a full binary install process for mac? I've tried docker, but the data returned is all off due to the runtime environment. A workable install and run process seems to have been explored here, but I'm concerned that it will not be fully compatible with the official methods/tutorials provided later on.

**Reply by luxiangzh** [reply]
> Actually, I took a chance and ran the agent with a nice 10 value, which seems to have been enough to place 'nice'.
> 
> To use this...
> 
> Download beszel agent
> 
> ```shell
> curl -sL "https://github.com/henrygd/beszel/releases/latest/download/beszel-agent_$(uname -s)_$(uname -m | sed -e 's/x86_64/amd64/' -e 's/armv6l/arm/' -e 's/armv7l/arm/' -e 's/aarch64/arm64/').tar.gz" | tar -xz -O beszel-agent | tee ./beszel-agent >/dev/null && chmod +x beszel-agent
> ```
> 
> Copy to /usr/local/bin
> 
> ```shell
> sudo cp beszel-agent /usr/local/bin
> ```
> 
> Create a LaunchDaemon
> 
> ```shell
> sudo nano /Library/LaunchDaemon/com.beszel.agent.plist
> ```
> 
> Edit the following, replacing the port and key as required
> 
> ```shell
> <?xml version="1.0" encoding="UTF-8"?>
> <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
> <plist version="1.0">
> <dict>
>     <key>Label</key>
>     <string>com.beszel.agent</string>
> 
>     <key>ProgramArguments</key>
>     <array>
>         <string>/bin/bash</string>
>         <string>-c</string>
>         <string>PORT=45876 KEY="ssh-ed25519 XXXXXX" /usr/local/bin/beszel-agent</string>
>     </array>
> 
>     <key>RunAtLoad</key>
>     <true/>
> 
>     <key>KeepAlive</key>
>     <dict>
>         
>         <key>SuccessfulExit</key>
>         <false/>
>         <key>ThrottleInterval</key>
>         <integer>60</integer> 
>     </dict>
> 
>     <key>StandardErrorPath</key>
>     <string>/var/log/beszel-agent.err</string>
>     <key>StandardOutPath</key>
>     <string>/var/log/beszel-agent.log</string>
> 
>     
>     <key>Nice</key>
>     <integer>10</integer>
> </dict>
> </plist>
> ```
> 
> Load the LaunchDaemon
> 
> ```shell
> sudo launchctl load /Library/LaunchDaemon/com.beszel.agent.plist
> ```
> 
> Test if it's running
> 
> ```shell
> pgrep -l beszel-agent
> 38277 beszel-agent
> ```

Just to help others, there's a small typo here where the folder name should be ```LaunchDaemons```

**Reply by henrygd** [reply]
Yes, it was brought up in #464 that we can use our GoReleaser config to make our git repository available as a homebrew tap.

Once we get that configured properly, we can just include the homebrew commands in the install script for easy setup.

https://goreleaser.com/customization/homebrew/

**Reply by Disin** [reply]
Hi, 
Any updates on this? I'm interested too.

**Reply by NightMean** [reply]
Thanks for this tutorial. I've decided to go with this route instead of going with homebrew solution below. 
However your steps didn't work for me directly. Not sure if something has changed between the MacOS versions (I'm a newbie to MacOS).
 
Here are the steps that worked for me one MacOS 13 - Ventura. 

1) Add new System in Bezsel Hub
2) Copy Token and Public key and press Add System
3) Download Bezsel Agent on MacOS: 
```bash
curl -sL "https://github.com/henrygd/beszel/releases/latest/download/beszel-agent_$(uname -s)_$(uname -m | sed -e 's/x86_64/amd64/' -e 's/armv6l/arm/' -e 's/armv7l/arm/' -e 's/aarch64/arm64/').tar.gz" | tar -xz -O beszel-agent | tee ./beszel-agent >/dev/null && chmod +x beszel-agent
```

4) Copy it to the /usr/local/bin
```bash
sudo cp beszel-agent /usr/local/bin
```

5) Create a plist file 
```
sudo nano /Library/LaunchDaemons/com.beszel.agent.plist
```

6) Copy & paste the config below to the terminal and adjust the `PORT`, `KEY`, `HUB_URL` and `TOKEN` variables below according to your setup

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
    <key>Label</key>
    <string>com.beszel.agent</string>

    <key>EnvironmentVariables</key>
    <dict>
        <key>PORT</key>
        <string>{YOUR_PORT_HERE}</string>
        
        <key>KEY</key>
        <string>ssh-ed25519 {YOUR_KEY_HERE}</string>
        
        <key>HUB_URL</key>
        <string>{YOUR_HUB_URL_HERE}</string>
        
        <key>TOKEN</key>
        <string>{YOUR_TOKEN_HERE}</string>
    </dict>

    <key>ProgramArguments</key>
    <array>
        <string>/usr/local/bin/beszel-agent</string>
    </array>

    <key>RunAtLoad</key>
    <true/>

    <key>KeepAlive</key>
    <dict>
        <key>SuccessfulExit</key>
        <false/>
        <key>ThrottleInterval</key>
        <integer>60</integer> 
    </dict>

    <key>StandardErrorPath</key>
    <string>/var/log/beszel-agent.err</string>
    
    <key>StandardOutPath</key>
    <string>/var/log/beszel-agent.log</string>

    <key>Nice</key>
    <integer>10</integer>
</dict>
</plist>
```

7) Press CTRL+O to save the config and CTRL+X to close nano

8) Launch bezsel agent
```bash
sudo launchctl bootstrap system /Library/LaunchDaemons/com.beszel.agent.plist
```

9) Check if everything is working correctly in log
```bash
cat /var/log/beszel-agent.err
```

10) Check if it's running (Not required as logs will show it as well)
```bash
pgrep -l beszel-agent
```

----

Optional troubleshooting steps  - In case you have something wrong in the config, use this command to terminate and unload it: 
```
sudo launchctl bootout system /Library/LaunchDaemons/com.beszel.agent.plist
```
Troubleshoot it by checking errors logs
```bash
cat /var/log/beszel-agent.err
```
Make changes in config with nano
```
sudo nano /Library/LaunchDaemons/com.beszel.agent.plist
```
Press CTRL+O to save the config and CTRL+X to close nano
Then reload it again with 
```bash
sudo launchctl bootstrap system /Library/LaunchDaemons/com.beszel.agent.plist
```

**Comment by henrygd** [comment]
Here's a command to install with Homebrew. You can add `-key "..."` at the end if you want.

```bash
curl -sL https://get.beszel.dev/brew -o /tmp/install-agent.sh && chmod +x /tmp/install-agent.sh && /tmp/install-agent.sh
```

Without install script:

```bash
mkdir -p ~/.config/beszel ~/.cache/beszel
echo 'KEY="ssh-ed25519 AAAA..."' > ~/.config/beszel/beszel-agent.env
brew tap henrygd/beszel
brew install beszel-agent
brew services start beszel-agent
```

Environment variables can be changed in `~/.config/beszel/beszel-agent.env`.

**Reply by bartvdbraak** [reply]
I did the above but I had to do `sudo brew services start beszel-agent`, 
```zsh
% sudo brew services start beszel-agent
Password:
Warning: Taking root:admin ownership of some beszel-agent paths:
  /opt/homebrew/Cellar/beszel-agent/0.10.2/bin
  /opt/homebrew/Cellar/beszel-agent/0.10.2/bin/beszel-agent-launcher
  /opt/homebrew/opt/beszel-agent
  /opt/homebrew/opt/beszel-agent/bin
  /opt/homebrew/var/homebrew/linked/beszel-agent
This will require manual removal of these paths using `sudo rm` on
brew upgrade/reinstall/uninstall.
Warning: beszel-agent must be run as non-root to start at user login!
==> Successfully started `beszel-agent` (label: homebrew.mxcl.beszel-agent)
```

otherwise I got:

```
Error: Permission denied @ rb_sysopen - /Users/<user>/Library/LaunchAgents/homebrew.mxcl.beszel-agent.plist
```

I have the logging directory and the env file:

```
<user>@<host> beszel % ls -la ~/.config/beszel
total 8
drwxr-xr-x  3 <user>  staff   96 Apr 18 11:23 .
drwxr-xr-x  4 <user>  staff  128 Apr 18 10:56 ..
-rw-r--r--  1 <user>  staff   98 Apr 18 10:56 beszel-agent.env
<user>@<host> beszel % cat ~/.config/beszel/beszel-agent.env
KEY="ssh-ed25519 AAAA<redacted>"
PORT=45876
<user>@<host> beszel % ls -la ~/.cache/beszel/
total 24
drwxr-xr-x  3 <user>  staff     96 Apr 18 11:09 .
drwxr-xr-x  5 <user>  staff    160 Apr 18 10:56 ..
-rw-r--r--  1 root     staff  11248 Apr 18 11:29 beszel-agent.log
```

But I only see this in the logs:

```
2025/04/18 11:28:41 Failed to load public keys:no key provided: must set -key flag, KEY env var, or KEY_FILE env var. Use 'beszel-agent help' for usage
```

**Reply by henrygd** [reply]
Thank you for testing!

That's a strange error.  I don't have much experience with homebrew, but my impression from the docs is that it's supposed to work at user-level, including services.

The key is missing because it's using `$HOME` to source `$HOME/.config/beszel/beszel-agent.env`. When run with sudo this means it's looking in the root user's home directory instead of your user's home directory where the file is.

The docs say to run `brew update` twice and `brew doctor` to display warnings about your installation.

You can also try uninstalling (`brew uninstall beszel-agent`) then fixing permissions before installing again:

```bash
sudo chown -R $(whoami) $(brew --prefix)/*
sudo chown -R $(whoami) ~/Library/LaunchAgents/
```

Reference: https://gist.github.com/irazasyed/7732946

**Reply by nephthys** [reply]
@henrygd Hi, It works much easier with homebrew than with docker-compose on my Mac Mini M4 (no need to use sudo for "brew services start" command in my case) ;) Would it be wise to add this installation method to the docs? It's already very complete, but this part is missing for macOS. In any case, your tool is GREAT, thank you and congratulations for the work.

**Reply by henrygd** [reply]
Thanks, the next release will be out soon with updated docs and an option to copy the brew command in the UI.

**Reply by miquelblanch** [reply]
I also have a Mac Mini M4 and I've already installed the agent using homebrew, works perfectly and I can perfectly monitor the server,... my problem is when trying to monitor my docker stats (installed with colima) in the same server,... I've tried to install the docker compose using portainer but seems that a server cannot be monitored twice, even we want to monitor different stats (the ones from the server, the other from docker...

In any case, thanks in advance for the support,...

**Reply by henrygd** [reply]
You don't need to run the agent with Docker to monitor Docker containers. You just need to tell the agent how to access the Docker API (usually with `docker.sock`).

On normal Linux installations `docker.sock` is available at `/var/run/docker.sock` so we can pick it up automatically.

I don't know anything about Colima, but I found this issue: https://github.com/abiosoft/colima/discussions/688

Looks like `docker.sock` is located at `/Users/myUser/.colima/default/docker.sock` by default.

Try adding this line to `~/.config/beszel/beszel-agent.env`, replacing `myUser` with your actual user:

```ini
DOCKER_HOST=/Users/myUser/.colima/default/docker.sock
```

Then restart the agent:

```bash
brew services restart beszel-agent
```

If this doesn't work, add `LOG_LEVEL=debug` to the env vars and restart the agent again.

Check the log output at the bottom of `~/.cache/beszel/beszel-agent.log` and please share any lines containing anything about Docker.

I'll look into it at some point but I have some higher priority things I need to work on at the moment.

**Reply by miquelblanch** [reply]
Hi, thank you very much for your quick response; really helpful cause it allows me to do a good troubleshooting.

Here's the thing:

• With the agent installed using colima (Docker), I can perfectly monitor all Docker containers (with volumes:  - /var/run/docker.sock:/var/run/docker.sock:ro).
• With the binary installed in the Mac Mini, I can perfectly monitor the server.

My problem is that I would like to have both simultaneously in the same instance because they are in the same server (same IP adress and the same Public Key).

• I’ve tried to include the “DOCKER_HOST=/Users/myUser/.colima/default/docker.sock” in the ~/.config/beszel/beszel-agent.env, but then the server went down.
• I’ve also tried to include the “DOCKER_HOST=/var/run/docker.sock” in the ~/.config/beszel/beszel-agent.env, but again the server went down.

Finally, I’ve included the “LOG_LEVEL=debug” in the file ~/.config/beszel/beszel-agent.env, but no logs were generated in the ~/.cache/beszel/beszel-agent.log file.

So in conclusion, the .env file should contain the following right?

DOCKER_HOST=/Users/myUser/.colima/default/docker.sock
KEY="ssh-ed25519 ……./"
LOG_LEVEL=debug

Seems that the solution goes to include the correct var DOCKER_HOST in the .env file but we need to find it.

Again thanks for all!

**Reply by henrygd** [reply]
No worries, the env file looks good assuming you replaced `myUser` with your actual username and have the full `KEY` value.

There should definitely be a way to make this work. Let's troubleshoot this step by step:  

### Socket Path Verification

Confirm the Docker socket exists at `/Users/<your-user>/.colima/default/docker.sock` by running:  

```bash  
ls /Users/<your-user>/.colima/default/ 
```  

Does it show `docker.sock`? Make sure you substitute `<your-user>` with your actual username. You can get this by running `whoami`.

### Monitoring Logs

Open a new terminal and run the following command to monitor the agent logs in real-time:

```bash
tail -f ~/.cache/beszel/beszel-agent.log
```

If the server goes down after setting `DOCKER_HOST`, review the logs in the terminal to figure out the exact cause.

### Colima-Specific Caveats

Since Colima runs Docker in a VM, the host socket path (`/Users/<your-user>/.colima/default/docker.sock`) may not work directly as it does on Linux.

You can test socket accessibility from the host using curl:  

```bash  
curl --unix-socket ~/.colima/default/docker.sock http://localhost/version  
```  

A JSON response with Docker version details confirms the socket is functional.  

### Alternative Workaround: Docker Socket Proxy

As a workaround, run [Tecnativa/docker-socket-proxy](https://github.com/Tecnativa/docker-socket-proxy) inside Colima to expose the Docker API on localhost over TCP:  

```bash  
docker run -d \  
  --name docker-proxy \  
  -v /var/run/docker.sock:/var/run/docker.sock \  
  -p 127.0.0.1:2375:2375 \  
  -e "CONTAINERS=1" \
  tecnativa/docker-socket-proxy  
```  

Then use this for the agent's `DOCKER_HOST`:  

```env  
DOCKER_HOST=tcp://127.0.0.1:2375  
```  

This avoids direct socket access and works around Colima's possible VM limitations. It's also good security practice because it restricts access to only what's needed.

If this doesn't work, please share any error messages from `~/.cache/beszel/beszel-agent.log` after restarting the service.

**Reply by miquelblanch** [reply]
Again thanks for your help. I found the solution!

The workaround you proposed me gave me the clue to find it. In fact the workaround worked and the tcp in front of the DOCKER_HOST made me finding the solution. The correct line is:

`DOCKER_HOST=unix:///Users/myUser/.colima/default/docker.sock`

JFYI, the error found in the logs i the previous DOCKER HOST format was: 

2025/05/05 08:20:56 ERROR Invalid DOCKER_HOST scheme=""

Thanks again for all!!

**Reply by henrygd** [reply]
No worries, I totally forgot that `DOCKER_HOST` sockets need the URL scheme. Glad it's working now! :+1:

**Comment by hype8912** [comment]
Having issues with Intel based Mac computers. I have a Mac Mini 2012 and a Macbook Pro 2019. I can't get Beszel to report anything. It's working fine on my 2 Mac Mini M series computers. Any tips to troubleshoot the issue or know if the Intel Macs are just a limitation?

**Reply by henrygd** [reply]
What versions of macOS? Go 1.26 requires 12 Monterey or later (and Go 1.27 will require 13 Ventura).

Do you have any logs in `~/.cache/beszel/beszel-agent.log`?

Try running directly and let me know what it outputs:

```bash
beszel-agent --key="..." --token="..." --url="http://localhost:8090"
```

I don't have a Mac, but from previous testing and user reports I would expect the 2019 Macbook to work.

**Reply by hype8912** [reply]
Both computers are running Sequoia. I did some troubleshooting last night and this morning. Running it from launchd with a plist file doesn't work but running it from the terminal works. As a workaround I added the beszel-agent-launcher to the login apps list and it works. I run the Mac Mini 2012 headless so it's not an issue to run it that way but on my MacBook I have to remember not to close the window. I'll do some more testing this week.

**Comment by pro-sumer** [comment]
Thanks for creating Beszel!

I have set it up on multiple systems, including my MacBook Air. On all other systems (Linux based) there is documentation on how to automatically update the Beszel agent, but I could not find such functionality for macOS.

How do Mac users make sure their agent is automatically updated?
